worker/src/cache.rs

```diff
@@ -47,6 +50,44 @@ impl Default for Cache {
     }
 }
 
+// Helper function to validate the headers of a Request or Response.
+// The cache API does not support 'stale-while-revalidate' and 'stale-if-error' values for cache-control directives.
+// More importantly, the cache API does NOT cache if the Response to be cached is missing a 'cache-control' header
+// with 'max-age' or 's-maxage' directives.
+fn check_headers(headers: &Headers, is_response: bool) -> Result<()> {
+    let cache_control_header = headers.get("cache-control");
+    match cache_control_header {
+        Ok(res) => match res {
+            Some(directive_value) => {
+                if directive_value == "stale-while-revalidate"
+                    || directive_value == "stale-if-error"
+                {
+                    return Err(Error::CacheControlHeader("'stale-while-revalidate' and 'stale-if-error' directives are not supported by the Cache API".to_string()));
+                }
+
+                if is_response {
```

Comment:
This helper function is used to check headers in `Cache::put` and `Cache::get`. 

Not sure if we should keep this. Looking at https://developers.cloudflare.com/cache/about/cache-control/ there can be other behaviors that do no necessarily use `max-age` and `s-maxage` directives. We could maybe remove this and instead point to documentations such as https://developers.cloudflare.com/workers/runtime-apis/cache/ and https://developers.cloudflare.com/cache/about/cache-control/?